.github/workflows/terraform-remote-state.yml

```diff
@@ -0,0 +1,23 @@
+name: Terraform remote state
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  terraform:
+    runs-on: ubuntu-latest
+
+    env:
+      ARM_CLIENT_ID: ${{ secrets.ARM_CLIENT_ID }}
+      ARM_CLIENT_SECRET: ${{ secrets.ARM_CLIENT_SECRET }}
+      ARM_SUBSCRIPTION_ID: ${{ secrets.ARM_SUBSCRIPTION_ID }}
+      ARM_TENANT_ID: ${{ secrets.ARM_TENANT_ID }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Terraform Init
+        working-directory: ./terraform/terraform-remote-state
+        run: terraform init
```

Comment:
does the GitHub agent have `terraform` installed as a command-line tool? You might need 

> - uses: hashicorp/setup-terraform@v1

https://github.com/hashicorp/setup-terraform#usage